pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -2527,13 +2531,12 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 					{
 						Type:   hivev1.ProvisionStoppedCondition,
 						Status: corev1.ConditionTrue,
-						Reason: "InstallAttemptsLimitReached",
 					},
 					{
 						Type:    hivev1.ProvisionedCondition,
-						Status:  corev1.ConditionFalse,
-						Reason:  hivev1.ProvisionedReasonProvisionStopped,
-						Message: "Provisioning failed terminally (see the ProvisionStopped condition for details)",
+						Status:  corev1.ConditionUnknown,
+						Reason:  hivev1.InitializedConditionReason,
+						Message: "Condition Initialized",
```

Comment:
The `Initialized` family should only be used when initializing the condition array at the top of the controller. None of those values should be used for any actual state. If we can't assert a specific actual state, better to leave the condition alone.

How do we get here? (I can take a closer look tomorrow.)

Reply:
This is the test case where the operator using the ClusterInstall crd is misbehaving and reporting erroneous conditions (specifically, stopped but not completed or failed).  I assume that provisioned condition is still unknown from when it was initially at the very beginning of the process.

Reply:
Mm, interesting. On the one hand, leaving the Provisioned condition as unknown/initialized seems reasonable, because we don't actually have any further information to impart. On the other hand, we _do_ know that they're violating the contract, and that's something we could potentially call out. In situations like this I'm tempted to make the user-facing message indicate that they should open a bug. If we were doing that, I guess it would be something like:

```go
						Type:    hivev1.ProvisionedCondition,
						Status:  corev1.ConditionUnknown,
						Reason:  "Error",  // This is what shows up in `oc get cd` in the `PROVISIONSTATUS` column
						Message: "Invalid ClusterInstall conditions. Please report this bug.",
```

As to which of these things we ought to do... I'm not sure. Let's bring it up after scrum, eh?